### PR TITLE
Define limit for negative rational powers (#11672)

### PR DESCRIPTION
**Pull request by apalmer1377**
When calculating the limit of a function, doit() now checks to see if the expression is a simple power sequence that converges to 0 (i.e., the limit referenced in #11672), and if it is, returns 0; otherwise, the algorithm continues as normal.

### CODE REVIEW

File: sympy/series/limits.py [hunk 1]
```diff
@@ -168,6 +168,16 @@ def doit(self, **hints):
         if e.is_Order:
             return Order(limit(e.expr, z, z0), *e.args[1:])
 
+        #Check to see if e is a simple power series, like (-1/2)**x
+        if e.is_Pow and len(e.free_symbols) == 1:
+            k = e.free_symbols.pop()
+            if k == z:
```
Comment (asmeurer) [inline sympy/series/limits.py]: Or rather `e.exp`, sorry.

**Comment by asmeurer** [comment]
This looks good, aside from those comments. There are other related cases like `limit(3**x, x, oo)`, but it seems that limit already handles those quite well.

File: sympy/series/limits.py (hunk 1 above)
Comment (apalmer1377) [inline sympy/series/limits.py]: Yeah, that was poorly thought out--I'll rewrite to only accept positive multiples of `z`.

File: sympy/series/limits.py [hunk 2]
```diff
@@ -168,6 +168,16 @@ def doit(self, **hints):
...
+                try:
+                    if (abs(e.base) < 1 and z0 is S.Infinity) or (abs(e.base) > 1 and z0 is S.NegativeInfinity):
```
Comment (apalmer1377) [inline sympy/series/limits.py]: That's what the `try` statement does--if `z` or any non-`Number` expression is in the base, then trying to assess the truth of `abs(e.base) < 1' throws a `TypeError`, and the function just continues on.

Comment (asmeurer) [inline sympy/series/limits.py]: I think it's possible for it to still happen that `abs(e.base) < 1` when `e.base` is symbolic. It's better to check this explicitly (`not e.base.has(z)`).

Comment (apalmer1377) [inline sympy/series/limits.py]: That makes sense--I'll add the test.

**Comment by moorepants** [comment]
This seems good to merge. @asmeurer please do so if you are good with the fixes.

**Comment by czgdp1807** [comment]
Closing in favour of https://github.com/sympy/sympy/pull/17939

**Pull request closed without merging**